Potentially Crucial Custom Rules Bug

# Issue Description

Disabling just one custom rule, actually disables all custom rules.

# STR

### 1) SwiftLint Config

Create config file named `.swiftlint.yml` with the following contents:

```yaml
custom_rules:
  hello_rule:
    regex: "Hello"
    match_kinds: string
  world_rule:
    regex: "World"
    match_kinds: string
```

### 2) Swift Test File

Create test file named `Example.swift` with the following contents:

```swift
class Example {

    func example() -> String {
        return "Hello World!"
    }

}
```

### 3) Run SwiftLint

When running SwiftLint, we successfully get the following output:

```
Loading configuration from '.swiftlint.yml'
Linting Swift files in current working directory
Linting 'example.swift' (1/1)
/PATH_OBSCURED/example.swift:4:17: warning: hello_rule Violation: Regex matched. (hello_rule)
/PATH_OBSCURED/example.swift:4:23: warning: world_rule Violation: Regex matched. (world_rule)
Done linting! Found 2 violations, 0 serious in 1 file.
```

As you can see ☝️ both rules are violated.

### 4) Disable One Rule

In the test file, disable only the `world_rule`:

```swift
class Example {

    // swiftlint:disable world_rule
    func example() -> String {
        return "Hello World!"
    }

}
```

### 5) Run SwiftLint Again

When running SwiftLint now, we get the following output:

```
Loading configuration from '.swiftlint.yml'
Linting Swift files in current working directory
Linting 'example.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

As you can see ☝️ there are no violations, when in fact, the `hello_rule` violation should be reported.

## Additional Info

This has been tested with various scenarios and the experience is consistently that disabling one custom rule, actually disables all custom rules. This also occurs with `:previous`, `:this`, `:next`.

This issue appears to only affects custom rules, not built-in rules.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -63,6 +63,9 @@
   [#614](https://github.com/realm/SwiftLint/issues/614)
   [#977](https://github.com/realm/SwiftLint/issues/977)
   [#1614](https://github.com/realm/SwiftLint/issues/1614)
+* Fix custom rules not working correctly with comment commands.  
+  [JP Simard](https://github.com/jpsim)
+  [#1558](https://github.com/realm/SwiftLint/issues/1558)
 
 * Fix incorrectly using configuration files named `.swiftlint.yml` when they are
   located in the same directory as a differently-named, user-provided custom

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -63,6 +63,7 @@
   [#614](https://github.com/realm/SwiftLint/issues/614)
   [#977](https://github.com/realm/SwiftLint/issues/977)
   [#1614](https://github.com/realm/SwiftLint/issues/1614)
+
 * Fix custom rules not working correctly with comment commands.  
   [JP Simard](https://github.com/jpsim)
   [#1558](https://github.com/realm/SwiftLint/issues/1558)

--- a/Source/SwiftLintFramework/Rules/CustomRules.swift
+++ b/Source/SwiftLintFramework/Rules/CustomRules.swift
@@ -101,9 +101,7 @@ public struct CustomRules: Rule, ConfigurationProviderRule, CacheDescriptionProv
                     return true
                 }
 
-                return !configurations.contains { config in
-                    return region.isRuleDisabled(customRuleIdentifier: config.identifier)
-                }
+                return !region.isRuleDisabled(customRuleIdentifier: configuration.identifier)
             }
         }
     }

--- a/Tests/SwiftLintFrameworkTests/CustomRulesTests.swift
+++ b/Tests/SwiftLintFrameworkTests/CustomRulesTests.swift
@@ -61,6 +61,16 @@ class CustomRulesTests: XCTestCase {
         XCTAssertEqual(customRules.validate(file: file), [])
     }
 
+    func testLocalDisableCustomRuleWithMultipleRules() {
+        let (configs, customRules) = getCustomRulesWithTwoRules()
+        let file = File(contents: "//swiftlint:disable \(configs.1.identifier) \n// file with a pattern")
+        XCTAssertEqual(customRules.validate(file: file),
+                       [StyleViolation(ruleDescription: configs.0.description,
+                                       severity: .warning,
+                                       location: Location(file: nil, line: 2, character: 16),
+                                       reason: configs.0.message)])
+    }
+
     func testCustomRulesIncludedDefault() {
         // Violation detected when included is omitted.
         let (_, customRules) = getCustomRules()
@@ -90,7 +100,7 @@ class CustomRulesTests: XCTestCase {
         XCTAssertEqual(violations.count, 0)
     }
 
-    func getCustomRules(_ extraConfig: [String:String] = [:]) -> (RegexConfiguration, CustomRules) {
+    private func getCustomRules(_ extraConfig: [String:String] = [:]) -> (RegexConfiguration, CustomRules) {
         var config = ["regex": "pattern",
                       "match_kinds": "comment"]
         extraConfig.forEach { config[$0] = $1 }
@@ -110,7 +120,36 @@ class CustomRulesTests: XCTestCase {
         return (regexConfig, customRules)
     }
 
-    func getTestTextFile() -> File {
+    private func getCustomRulesWithTwoRules() -> ((RegexConfiguration, RegexConfiguration), CustomRules) {
+        let config1 = ["regex": "pattern",
+                      "match_kinds": "comment"]
+
+        var regexConfig1 = RegexConfiguration(identifier: "custom1")
+        do {
+            try regexConfig1.apply(configuration: config1)
+        } catch {
+            XCTFail("Failed regex config")
+        }
+
+        let config2 = ["regex": "something",
+                       "match_kinds": "comment"]
+
+        var regexConfig2 = RegexConfiguration(identifier: "custom2")
+        do {
+            try regexConfig2.apply(configuration: config2)
+        } catch {
+            XCTFail("Failed regex config")
+        }
+
+        var customRuleConfiguration = CustomRulesConfiguration()
+        customRuleConfiguration.customRuleConfigurations = [regexConfig1, regexConfig2]
+
+        var customRules = CustomRules()
+        customRules.configuration = customRuleConfiguration
+        return ((regexConfig1, regexConfig2), customRules)
+    }
+
+    private func getTestTextFile() -> File {
         return File(path: "\(bundlePath)/test.txt")!
     }
 }

